sql/files/examples/boolfind_cmp/check_boolfind.c

```diff
@@ -48,7 +48,7 @@ int main(int argc, char **argv)
 
 		outputvalid = 1;
 		if ( strncmp(line,"OUTPUT ",7)!=0 ||
-		     strlen(line)<7 || line[7]=='0' ) outputvalid = 0;
+		     strlen(line)<7 || line[7]==0 ) outputvalid = 0;
```

Comment:
The condition `line[7]=='0'` was meant to check that the number does not have prefix zeros. It does this incorrectly, but the proposed change also isn't effective: if `strlen(line)<7` is not true then `line[7]==0` is also guaranteed to not be true, so the condition is superfluous. I think this line should be changed to:
```c
    (strlen(line)>=8 && line[7]=='0') ) outputvalid = 0;
```